packages/core/custom-branding/core-custom-branding-server-internal/custom_branding_service.ts

```diff
@@ -10,13 +10,12 @@ import type { CustomBranding } from '@kbn/core-custom-branding-common';
 import type { KibanaRequest } from '@kbn/core-http-server';
 import type { CoreContext } from '@kbn/core-base-server-internal';
 import type { Logger } from '@kbn/logging';
-
 /**
  * @internal
  */
 export interface InternalCustomBrandingSetup {
   register: (pluginName: string, fetchFn: CustomBrandingFetchFn) => void;
-  getBrandingFor: (request: KibanaRequest) => Promise<CustomBranding>;
+  getBrandingFor: (request: KibanaRequest, unauthenticated?: boolean) => Promise<CustomBranding>;
```

Comment:
NIT: if we go with adding this `unauthenticated` option, it probably shouldn't be optional in the API definition. (no `?`)